### PR TITLE
feat: made x/tokenfactory mint and burn methods public

### PR DESCRIPTION
**Pull request by keruch**
Closes: #7670

Duplicate of https://github.com/osmosis-labs/osmosis/pull/7671

## What is the purpose of the change

I made x/tokenfactory mint and burn methods public to allow other modules to use these methods through Go interfaces.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

File: x/tokenfactory/keeper/bankactions.go [hunk 1]
```diff
@@ -2,10 +2,35 @@ package keeper
...
+	if sender != authorityMetadata.GetAdmin() {
+		return types.ErrUnauthorized
+	}
+
+	err = k.mintTo(ctx, amount, mintTo)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (Pipello) [inline x/tokenfactory/keeper/bankactions.go:31]: nit: 
```suggestion
	return k.mintTo(ctx, amount, mintTo)
```

File: x/tokenfactory/keeper/bankactions.go [hunk 2]
```diff
@@ -28,6 +53,30 @@ func (k Keeper) mintTo(ctx sdk.Context, amount sdk.Coin, mintTo string) error {
...
+	if ok {
+		return types.ErrBurnFromModuleAccount
+	}
+
+	err = k.burnFrom(ctx, amount, burnFrom)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (Pipello) [inline x/tokenfactory/keeper/bankactions.go:77]: nit: 
```suggestion
	return k.burnFrom(ctx, amount, burnFrom)
```

File: x/tokenfactory/keeper/bankactions.go (hunk 1 above)
Comment (keruch) [reply x/tokenfactory/keeper/bankactions.go:31]: I did this intentionally to make error handling as explicit as possible. Additionally, this ensures that we won't forget to wrap this error if we decide to do so in the future.

File: x/tokenfactory/keeper/bankactions.go (hunk 2 above)
Comment (keruch) [reply x/tokenfactory/keeper/bankactions.go:77]: The same applies here

**Review by czarcas7ic — DISMISSED** [review]

**Pull request closed without merging**